### PR TITLE
feat(ts): transpile each plugin separately

### PR DESCRIPTION
**Pull request by schultzp2020**
Transpiles each plugin separately to take advantage of `turbo`.

Required for https://github.com/janus-idp/backstage-plugins/pull/618 because the nexus plugin requires a script to run before `tsc`.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,14 +0,0 @@
-{
```
Comment (schultzp2020) [inline tsconfig.json:1]: No longer needed because each package/plugin looks locally for `tsconfig.json`

File: turbo.json [hunk 2]
```diff
@@ -5,30 +5,22 @@
...
     },
     "lint": {},
-    "//#tsc": {
-      "outputs": ["dist-types/**"]
+    "tsc": {
+      "outputs": ["../../dist-types/**"],
+      "dependsOn": ["^tsc"]
     }
   },
-  "globalDependencies": ["tsconfig.json"]
```
Comment (schultzp2020) [inline turbo.json:33]: No more global `tsconfig.json`

File: packages/app/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "@backstage/cli/config/tsconfig.json",
+  "include": ["src"],
+  "exclude": ["node_modules"],
+  "compilerOptions": {
+    "outDir": "../../dist-types/packages/app",
```
Comment (schultzp2020) [inline packages/app/tsconfig.json:6]: Sadly `yarn build` (i.e. `backstage-cli package build`) requires `dist-types` to be at the root level.

**Review by invincibleJai — APPROVED** [review]
LGTM

**Pull request merged**